### PR TITLE
Selenium: cover Yaml language server functionality by selenium test

### PR DESCRIPTION
**Pull request by SkorikSergey**
### What does this PR do?
This PR creates **YamlFileEditingTest** selenium test that cover the **Yaml** language server functionality by use cases from **Language servers** test plan - https://github.com/eclipse/che/wiki/Test-plan-for-checking-Language-servers.

### What issues does this PR fix or reference?
https://github.com/eclipse/che/issues/10442

### CODE REVIEW

**Comment by SkorikSergey** [comment]
ci-test

**Comment by riuvshin** [comment]
ci-test build report:
[Build details](https://ci.codenvycorp.com/job/che-pullrequests-test-ocp/512/)
[Test report](https://ci.codenvycorp.com/job/che-pullrequests-test-ocp/512//Selenium_tests_report)
[selenium tests report data](http://tests.codenvycorp.com/selenium_test_report_data/10525)
docker image: eclipseche/che-server:10525
https://github.com/orgs/eclipse/teams/eclipse-che-qa please check this report.

File: selenium/che-selenium-test/src/main/java/org/eclipse/che/selenium/pageobject/CodenvyEditor.java [hunk 1]
```diff
@@ -2167,4 +2172,13 @@ public void moveCursorToText(String text) {
...
         By.xpath(format(Locators.TEXT_TO_MOVE_CURSOR_XPATH, text)));
   }
+
+  public void checkProposalDocumentation(String expectedText) {
+    seleniumWebDriverHelper.waitTextContains(proposalDoc, expectedText);
+  }
+
+  public void launchCommentCodeFeature() {
+    Actions action = actionsFactory.createAction(seleniumWebDriver);
+    action.keyDown(CONTROL).sendKeys("/").keyUp(CONTROL).perform();
```
Comment (Ohrimenko1988) [inline selenium/che-selenium-test/src/main/java/org/eclipse/che/selenium/pageobject/CodenvyEditor.java]: You may simplify:
```
actionsFactory.createAction(seleniumWebDriver)
    .keyDown(CONTROL).sendKeys("/").keyUp(CONTROL).perform();
```

File: selenium/che-selenium-test/src/main/java/org/eclipse/che/selenium/pageobject/Preferences.java [hunk 2]
```diff
@@ -311,6 +323,21 @@ public void clickOnGenerateKeyButton() {
         .click();
   }
 
+  public void clickOnAddSchemaUrlButton() {
```
Comment (Ohrimenko1988) [inline selenium/che-selenium-test/src/main/java/org/eclipse/che/selenium/pageobject/Preferences.java:326]: please use "waitAndClick" method from "SeleniumWebDriverHelper"

File: selenium/che-selenium-test/src/main/java/org/eclipse/che/selenium/pageobject/Preferences.java [hunk 3]
```diff
@@ -311,6 +323,21 @@ public void clickOnGenerateKeyButton() {
...
+    new WebDriverWait(seleniumWebDriver, LOAD_PAGE_TIMEOUT_SEC)
+        .until(visibilityOf(addSchemaUrlButton))
+        .click();
+  }
+
+  public void addSchemaUrl(String schemaName) {
+    new WebDriverWait(seleniumWebDriver, LOAD_PAGE_TIMEOUT_SEC)
```
Comment (Ohrimenko1988) [inline selenium/che-selenium-test/src/main/java/org/eclipse/che/selenium/pageobject/Preferences.java]: Use "SeleniumWebDriverHelper"

File: selenium/che-selenium-test/src/main/java/org/eclipse/che/selenium/pageobject/Preferences.java [hunk 4]
```diff
@@ -311,6 +323,21 @@ public void clickOnGenerateKeyButton() {
...
+        .until(visibilityOf(addSchemaUrlInput));
+    addSchemaUrlInput.sendKeys(schemaName);
+    new WebDriverWait(seleniumWebDriver, LOAD_PAGE_TIMEOUT_SEC)
```
Comment (Ohrimenko1988) [inline selenium/che-selenium-test/src/main/java/org/eclipse/che/selenium/pageobject/Preferences.java]: Use "SeleniumWebDriverHelper"

File: selenium/che-selenium-test/src/test/java/org/eclipse/che/selenium/languageserver/YamlFileEditingTest.java [hunk 5]
```diff
@@ -0,0 +1,224 @@
...
+    // comment lines by Ctrl+'/' buttons
+    editor.goToPosition(18, 1);
+    editor.launchCommentCodeFeature();
+    editor.waitTextIntoEditor("#  generation: 4");
+    editor.goToPosition(20, 1);
+    editor.launchCommentCodeFeature();
+    editor.goToPosition(22, 1);
+    editor.launchCommentCodeFeature();
+    editor.waitTextIntoEditor(
+        "#  generation: 4\n"
```
Comment (Ohrimenko1988) [inline selenium/che-selenium-test/src/test/java/org/eclipse/che/selenium/languageserver/YamlFileEditingTest.java]: constant ?

File: selenium/che-selenium-test/src/test/java/org/eclipse/che/selenium/languageserver/YamlFileEditingTest.java [hunk 6]
```diff
@@ -0,0 +1,224 @@
...
+
+    // uncomment all commented lines
+    editor.goToPosition(18, 1);
+    editor.launchCommentCodeFeature();
+    editor.goToPosition(20, 1);
+    editor.launchCommentCodeFeature();
+    editor.goToPosition(22, 1);
+    editor.launchCommentCodeFeature();
+    editor.waitTextIntoEditor(
+        "  generation: 4\n"
```
Comment (Ohrimenko1988) [inline selenium/che-selenium-test/src/test/java/org/eclipse/che/selenium/languageserver/YamlFileEditingTest.java]: constant ?

**Comment by Ohrimenko1988** [comment]
Don't forget to add the "YamlFileEditingTest" test to the suite xml.

**Review by Ohrimenko1988 — APPROVED** [review]

**Review by artaleks9 — APPROVED** [review]

**Pull request merged**